app/uk/gov/hmrc/slacknotifications/services/NotificationService.scala

```diff
@@ -63,24 +64,54 @@ class NotificationService @Inject()(
           sendSlackMessage(fromNotification(notificationRequest, slackChannel), service)
         }.map(flatten)
 
+
       case ChannelLookup.TeamsOfGithubUser(_, githubUsername) =>
         if (slackNotificationConfig.notRealGithubUsers.contains(githubUsername))
           Future.successful(NotificationResult().addExclusion(NotARealGithubUser(githubUsername)))
         else
           userManagementService.getTeamsForGithubUser(githubUsername).flatMap { allTeams =>
             withNonExcludedTeams(allTeams.map(_.team)) { nonExcludedTeams =>
               if (nonExcludedTeams.nonEmpty)
-                traverseFuturesSequentially(nonExcludedTeams) { teamName =>
-                  withExistingSlackChannel(teamName) { slackChannel =>
-                    sendSlackMessage(fromNotification(notificationRequest, slackChannel), service)
-                  }
-                }.map(flatten)
+                sendSlackMessageToTeams(nonExcludedTeams, notificationRequest, service)
               else
                 Future.successful(NotificationResult().addError(TeamsNotFoundForGithubUsername(githubUsername)))
             }
           }
+
+      case ChannelLookup.TeamsOfLdapUser(_, ldapUsername) =>
+        userManagementService.getTeamsForLdapUser(ldapUsername).flatMap { allTeams =>
+          withNonExcludedTeams(allTeams.map(_.team)) { nonExcludedTeams =>
+            if (nonExcludedTeams.nonEmpty)
+              sendSlackMessageToTeams(nonExcludedTeams, notificationRequest, service)
+            else {
+              sendSlackMessage(SlackMessage(
```

Comment:
How are missing teams reported for Github users? Are clients currently handling this? Can this be consistent?